src/providers/Youtube/Youtube.provider.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState, useContext } from 'react';
+
+import { API_URL } from '../../constants';
+
+const YoutubeContext = React.createContext(null);
+
+async function callYoutubeApi(search) {
+  const url = `${API_URL}/${search}`;
+  const response = await fetch(url);
+  const jsonResponse = await response.json();
+  return jsonResponse.items;
+}
+
+function useYoutube() {
+  const context = useContext(YoutubeContext);
+  if (!context) {
+    throw new Error(`Can't use "useYoutube" without an YoutubeContext!`);
+  }
+  return context;
+}
+
+function YoutubeProvider({ children }) {
```

Comment:
This works well and the api is very well structured but when creating a custom hook you have to ask yourself, is this state and functionality is going to be reusable by other components? With this, it seems to me that you took the responsibilities of a couple of components and bundled them inside a hook and then you only imported a subset of the hook's functionality into those components.

**state**: _video_ and **api-call**: _getVideo_ are used by VideoDisplay
**state**: _videos_ by VideoGrid
**api-call**: _searchVideos_ by Homepage

I suggest creating a module called **api** with the methods **getVideo**, **searchVideos**, **getVideo** and import them into the components that will need them; and let the components themselves manage their own state and call the methods when they need to.